scripts/build/termux_step_install_license.sh

```diff
@@ -1,87 +1,85 @@
+# shellcheck shell=bash
 termux_step_install_license() {
-	[ "$TERMUX_PKG_METAPACKAGE" = "true" ] && return
+	[[ "$TERMUX_PKG_METAPACKAGE" == 'true' ]] && return
 
 	mkdir -p "$TERMUX_PREFIX/share/doc/$TERMUX_PKG_NAME"
-	local LICENSE
-	local COUNTER=0
-	if [ ! "${TERMUX_PKG_LICENSE_FILE}" = "" ]; then
-		INSTALLED_LICENSES=()
+	local LICENSE COUNTER=0
+
+	# Was a license file specified?
+	if [[ -n "${TERMUX_PKG_LICENSE_FILE}" ]]; then
 		COUNTER=1
+		local INSTALLED_LICENSES=()
 		while read -r LICENSE; do
-			[ -z "${LICENSE}" ] && continue
-			if [ ! -f "$TERMUX_PKG_SRCDIR/$LICENSE" ]; then
+			# Skip empty lines
+			[[ -z "${LICENSE}" ]] && continue
+
+			# Check that the license file exists in the source files
+			[[ -f "$TERMUX_PKG_SRCDIR/$LICENSE" ]] || {
 				termux_error_exit "$TERMUX_PKG_SRCDIR/$LICENSE does not exist"
-			fi
-			if [[ " ${INSTALLED_LICENSES[@]} " =~ " $(basename $LICENSE) " ]]; then
+			}
+
+			if [[ " ${INSTALLED_LICENSES[*]} " == *" $(basename "$LICENSE") "* ]]; then
```

Comment:
Ensure `basename` of license does not contain a space itself.

Reply:
That would not matter based on the quoting.

Reply:
Yeah, I think that would trigger the earlier file not found error due to IFS splitting on space.

Reply:
There is no word splitting happening.
It's quoted.

Reply:
But generally, that check needs to be added, otherwise `" foo bar baz "` would match against ` " foo bar "` too.

Reply:
> But generally, that check needs to be added, otherwise `" foo bar baz "` would match against ` " foo bar "` too.

Nope
![image](https://github.com/user-attachments/assets/ee9af659-4805-4a39-86d6-fe9a6654641e)

Reply:
>There is no word splitting happening.

I meant in `read`.

Reply:
Except...
That isn't what we have, is it?
![image](https://github.com/user-attachments/assets/cc5cc5d5-a79c-44fe-9b2b-dd64ca645116)

Now I get what you meant.

Reply:
>Nope

You are missing the globs.

`[[ " foo bar baz " == *" foo bar "* ]] && echo true`

Reply:
>Now I get what you meant.

Yup. People generally don't add that check for the splitter itself, so can result in false positives.

Reply:
We could sidestep the issue by coverting `INSTALLED_LICENSES=()` to a hashmap, and matching on the keys.
Although that is admittedly slightly cursed.

Reply:
Can always check if `$TERMUX_PKG_LICENSE_FILE` contains a space before looping, that should keep current logic. The patch files containing a space are also missed due to similar issues, see https://github.com/termux/termux-packages/commit/2b27086129ef9d89f022bb482daf92e889f4b3da.

Reply:
That would need additional logic compared to a key lookup in an associative array.

Reply:
If referring to patch files, can be solved by running `find . -name "*.patch"` and grepping for a space or tab.

Reply:
I was thinking something along the lines of...
```bash
declare -A INSTALLED_LICENSES=()
# [...]
if (( ${#INSTALLED_LICENSES[$LICENSE_FILEPATH]:-} )); then
    # [...]
else
    # yes this is a valid way to assign key value pairs to a hashmap
    INSTALLED_LICENSES+=("${LICENSE_FILEPATH}" 'already installed')
fi
```
I might have understated when I said matching on hashmap keys was a slightly cursed approach.

Reply:
This should work if you wanna use associative array.

```shell
unset INSTALLED_LICENSES
local -A INSTALLED_LICENSES=()
local license_filename
while read -r LICENSE; do
    ...
    license_filename="$(basename "$LICENSE")"
    if [[ -n "${INSTALLED_LICENSES["$license_filename"]:-}" ]]; then
    	echo "already installed"
        continue
    fi
   
    INSTALLED_LICENSES["$license_filename"]=1
    # install now

done <<< "${TERMUX_PKG_LICENSE_FILE//,/$'\n'}"
```

Reply:
Okay we're on the same page then.
I think I will use your version, it looks slightly less cursed.

Reply:
Lolz, go ahead, and I am pretty sure glob pattern is much more cursed than associative array. This loop also works if you want a normal array at the end by maintaining a separate `local -a` array for it and appending to it if "already added" list doesn't have the element.

Reply:
`poppler-data` should be the perfect "smoke test" for this problem.
https://github.com/termux/termux-packages/blob/9b9e2b87f6c751958b7ef0a546d42756cb96002f/packages/poppler-data/build.sh#L1-L9